### PR TITLE
Automattic for Agencies: Redesign hosting marketplace page

### PR DESCRIPTION
**Pull request by yashwin**
Resolves https://github.com/Automattic/jetpack-roadmap/issues/1563

## Proposed Changes

This PR redesigns the hosting marketplace page

## Testing Instructions

1. Open the A4A live link.
2. Visit the Marketplace > Hosting page and verify the UI is redesigned as shown below and matches with the latest design (link on the ticket). Also, verify it works well on the screen sizes. 

screencapture-agencies-localhost-3000-marketplace-hosting-2024-05-20-17_22_42

Screenshot 2024-05-20 at 5 25 11 PM

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by keoshi — CHANGES_REQUESTED** [review]
Looks and works great, except for mobile. It doesn't seem to show the tooltip on tap.

Would it be possible to add that?

image

Thank you!

**Comment by yashwin** [comment]
Thanks for the review, @keoshi!

> Looks and works great, except for mobile. It doesn't seem to show the tooltip on tap. 
> Would it be possible to add that?

Yes, I added an `onTouchStart` event that should handle this.


https://github.com/Automattic/wp-calypso/assets/10586875/36786fb1-4b48-4a6f-a457-ba37b7eafd18

**Comment by jkguidaven** [comment]
The Migration offer card seems way too close to the hosting cards.

Screenshot 2024-05-17 at 7 21 54 PM

From Design
Screenshot 2024-05-17 at 7 23 19 PM

**Comment by jkguidaven** [comment]
I also noticed that the border-radius on the new design uses 8px. @keoshi is this correct?

File: client/a8c-for-agencies/style.scss [hunk 1]
```diff
@@ -414,3 +414,18 @@ html {
...
+
+.gray-tooltip {
+	.popover__arrow {
+		&::before {
+			border-bottom-color: var(--studio-gray-60) !important;
+			inset-block-start: 1px !important;
+		}
+	}
+	.popover__inner {
+		background: var(--studio-gray-60);
```
Comment (jkguidaven) [inline client/a8c-for-agencies/style.scss]: Can we replace the studio color reference here with theme properties?

File: client/a8c-for-agencies/style.scss [hunk 2]
```diff
@@ -414,3 +414,18 @@ html {
 		}
 	}
 }
+
+.gray-tooltip {
+	.popover__arrow {
+		&::before {
+			border-bottom-color: var(--studio-gray-60) !important;
```
Comment (jkguidaven) [inline client/a8c-for-agencies/style.scss]: If possible, can we avoid using `!important` here?

**Review by jkguidaven — COMMENTED** [review]
Thanks for working on this. Works great. I left few comments though.

**Comment by yashwin** [comment]
Thanks for the review, @jkguidaven!

I have addressed your comments: https://github.com/Automattic/wp-calypso/pull/90837/commits/27fb2ce4b2025c18079f5ac4e8a59e387e6e3a5d

File: client/a8c-for-agencies/style.scss (hunk 1 above)
Comment (yashwin) [reply client/a8c-for-agencies/style.scss]: Sorry, I copied this code from another place, so I missed this.

File: client/a8c-for-agencies/style.scss (hunk 2 above)
Comment (yashwin) [reply client/a8c-for-agencies/style.scss]: Sorry, I copied this code from another place, so I missed this.

**Review by jkguidaven — APPROVED** [review]
Code wise this LGTM. Changes also works great as per test instructions. 👍

**Review by keoshi — APPROVED** [review]
Looks great to me, Yashwin! Have two small comments that are non-blocking, so I'm approving the PR.

#### Doesn't respond to simple tap

On mobile, it seems like the tooltip is only shown when tap + moving the tap point is involved. In other others, it doesn't seem to respond to a simple tap on the same spot.

I'm testing this in the laptop, simulating a touch device, so it might be that causing the issue. If we can't fix it on mobile, it might be worth to always display the tooltip on touch devices.

#### Shared benefit should be full-width

According to the latest designs by @jeffgolenski — https://www.figma.com/design/sX0uNGAgDUceP3nKu7iQ4z/Purchase-flows?node-id=8346-138635&t=X10Un6R2s1Trcn6F-11 — the bottom section that groups shared benefits should be full-width.

Also notice the title should be "Included with Standard & Premier hosting".

image

---

Thanks for all your hard work on this, @yashwin !

**Comment by yashwin** [comment]
Thanks for the review, @keoshi!

> #### Doesn't respond to simple tap

I think this should work on mobile devices. I'll do a round of testing on mobile once this PR is shipped and fix it if needed in another PR.

> #### Shared benefit should be full-width
> According to the latest designs by @jeffgolenski — https://www.figma.com/design/sX0uNGAgDUceP3nKu7iQ4z/Purchase-flows?node-id=8346-138635&t=X10Un6R2s1Trcn6F-11 — the bottom section that groups shared benefits should be full-width.

Thanks for linking the latest design.

> Also notice the title should be "Included with Standard & Premier hosting".

I have updated it now, including some text changes: https://github.com/Automattic/wp-calypso/pull/90837/commits/34f430d79a31fb07aa96c1c89f40f1fd840b460c

screencapture-agencies-localhost-3000-marketplace-hosting-2024-05-20-17_22_42

**Comment by keoshi** [comment]
Perrrfect — ship it! 🚢

**Comment by simonktnga8c** [comment]
Can we pause on shipping this one @yashwin please, sorry if it's a pain.

**Comment by simonktnga8c** [comment]
Want to get the final go-ahead from leadership before we do.

**Comment by simonktnga8c** [comment]
Thanks for reverting 👍 seeking clarity from leadership now. I'll let you know as soon as I do.

**Comment by yashwin** [comment]
@simonktnga8c: Sure, I have shipped the revert PR and created a new one now: https://github.com/Automattic/wp-calypso/pull/90890

**Pull request merged**